Samples ReadMe (#35)
realm/realm-dart-samples: flutter_flexible_sync/README.md

![Realm](https://github.com/realm/realm-dart/raw/main/logo.png)

[![License](https://img.shields.io/badge/License-Apache-blue.svg)](LICENSE)
# Flutter Flexible Sync sample
A simple application using the [Realm Flutter SDK](https://www.mongodb.com/docs/realm/sdk/flutter/) Flexible Sync with an [Atlas App Services application](https://www.mongodb.com/docs/atlas/app-services/).

> This sample demonstrates the usage of Flexible Sync. 

Reviewer: ```suggestion
```

Writing to a synced realm named `db_allTasks.realm` sends the data automatically to a linked MongoDB collection on Atlas.

> Then the data are downloaded back by the synchronization process to two separate realms,

Reviewer: No new line

Author: Added a new line before "Writing to a synced realm named"

> Then the data are downloaded back by the synchronization process to two separate realms,

Reviewer: ```suggestion
Then the data is downloaded back by the synchronization process to two separate realms,
```

> Then the data are downloaded back by the synchronization process to two separate realms,

Reviewer: ```suggestion
```

This seems confusing. Delete it imo

Reviewer: I am not sure why we are doing this on this sample. But we should reword this somewhat. 

So perhaps

```
Writing to a synced realm named db_allTasks.realm syncs the data automatically to a linked MongoDB collection on Atlas. 
Then sample demonstrates working with two subscription queries for important tasks and normal tasks in two different realms.

Author: OK.

`db_importantTasks.realm` and  `db_normalTaks.realm`, filtered by specific subscription query.

## Realm Flutter SDK 

Realm Flutter package is published to [realm](https://pub.dev/packages/realm).

## Environment

> * Realm Flutter supports the platforms iOS, Android, Windows, MacOS and Linux.

Reviewer: ```suggestion
* Realm Flutter supports iOS, Android, Windows, MacOS and Linux platforms.
```

* Flutter ^3.0.3 or newer
* For Flutter Desktop environment setup, see [Desktop support for Flutter](https://docs.flutter.dev/desktop).

## Atlas App Services Configuration Steps

> This sample is using an already prepared Atlas App Service with AppID `flutter_flexible_sync-rmjux`.

Reviewer: ```suggestion
This sample is using an existing Atlas App Service with AppID `flutter_flexible_sync-rmjux`.
```

> If you want to create your own Atlas App Service and to have an access to the cloud App, follow the instruction below.

Reviewer: ```suggestion
If you want to create an Atlas App Service and have an access to the cloud App, follow the instruction below.
```

    For the purpose of this sample you don't need to create an app from a template. You can just create an empty application.
1. Go to the **Authentication** menu in the left panel and make sure the option "Allow users to log in anonymously" under `Authentication providers` tab is ON. Save and then click the button in the blue line above - `Review draft & deploy`. Read [Authentication Providers](https://www.mongodb.com/docs/atlas/app-services/authentication/providers/) for more information about the other authentication types.
1. Go to the **Device Sync** menu and [Enable Flexible Sync](https://www.mongodb.com/docs/atlas/app-services/sync/configure/enable-sync/#enable-flexible-sync).
    * Don't create a schema. Skip it by choosing "No thanks, continue to Sync".
    * Press the "Flexible Sync" button. Only Flexible Sync is supported in the Realm Flutter SDK.
    * Switch ON the ["Development mode"](https://www.mongodb.com/docs/atlas/app-services/sync/data-model/development-mode/) option.
    * Create a new database collection and choose a name for it.

>     * Create a new queryable field used for filtering data between both realms in this application. For our sample the field is `isImportant`.

Reviewer: ```suggestion
    * Create a new queryable field used for filtering data between both realms in this application. For this sample the field is `isImportant`.
```
